doc/logging_monitoring.md

```diff
@@ -0,0 +1,197 @@
+files and buffers them in memory (filtering them for the right service(s) if needed) until they are requested. The number of lines buffered can be specified as well as a lifespan for the log session.
+
+# Monitoring
+
+The base class contains a basic monitoring solution which saves message traces to a file.
+
+## Initialization
+
+By default monitoring messages of the whole pipeline get logged in a `monitoring.log` file. The filename can be set manually by specifying the monitoring field in the service initialization similar to the example below.
+
+```json
+{
+  "name": "myservice_instance",
+  "ip": "127.0.0.1",
+  "port": 1337,
+  "logging": {
+      "logpath": "/path/to/file.log",
+      "level": "info"
+  },
+  "monitoring": {
+      "filename": "/path/to/monitoring/file.log"
+  }
+}
+```
+
+To overwrite the default values you might add default parameters for logging in the roxconnector plugin parameter configs (similar to default logging params):
+
+```json
+"default": {
+  "monitoring": {
+    "filename": "monitoring.log",
+    "monitor_class": "roxcomposer.monitor.basic_monitoring.BasicMonitoring"
+  }
+}
+```
+
+## Monitoring messages
+
+There are several situations where the basic monitoring creates messages. Currently the following events trigger a monitoring message automatically:
+
+* a ROXcomposerMessage was received
+* a ROXcomposerMessage was dispatched
+* a ROXcomposerMessage reached it's final destination
+
+If everything works as expected a monitoring message like the following should be generated:
+
+```bash
+{"event": "message_dispatched", "status": "in_transit", "time": 1513596460.0961697, "args": {"service_name": "service1", "message_id": "dfa156e3-a8f6-4968-a255-ebd44e41d846", "destination": "127.0.0.1:10000"}}
+```
+
+Furthermore, the basic monitoring can be used to report an error state for a message:
+
+```python
+self.monitoring.msg_error(
+    service_name=self.params['name'],
+    message_id="dfa156e3-a8f6-4968-a255-ebd44e41d846",
+    description=errmsg
+)
+```
+
+In addition custom metrics can be commited to the monitoring file via the `custom_metric` function from the basic monitoring class.
+The function expects a service name, a metric name and a metric dictionary as keyword arguments.
+Please consider the following example.
+
+```python
+custom = {"service_name": "serv1", "metric_name": "some-metric", "metric_dictionary":
+                {"metric": "entry", "metric2": "another-entry"}}
+monitor.custom_metric(**custom)
+```
+
+## Reporting
+
+The basic monitoring class has a counterpart named `BasicReporting`. This class holds the functions to retrieve monitoring information on specific events. Either their whole monitoring history or only the last
+known status. See the API [documentation](rest.md) for details on accessing the reporting functionionality.
+
+# Logging injection
+
+Any logging class needs to adhere to the following interface to be compatible to the base service:
```

Comment:
Something went wrong here, the text seems to be redundant.
Because of the ':' it looks like a part is missing, so maybe leave it out

Reply:
Fixed.